### PR TITLE
Add option for ingress classes

### LINKED ISSUE #663: add option to sync ingress classes
If ingress syncing is enabled, it would be great if we could sync ingress classes as well from the host to the virtual cluster. Should work similar to legacy-storageclasses syncer.

### PR DESCRIPTION
**Pull request by ishankhare07**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #663 

**Please provide a short message that should be published in the vcluster release notes**
Added option to vcluster to sync `ingressclasses` if `ingress` sync is enabled

**What else do we need to know?**

### CODE REVIEW

File: cmd/vcluster/context/context.go [hunk 1]
```diff
@@ -110,6 +110,7 @@ var ExistingControllers = map[string]bool{
 	"fake-persistentvolumes": true,
 	"persistentvolumeclaims": true,
 	"ingresses":              true,
```
Comment (FabianKramm) [inline cmd/vcluster/context/context.go:112]: Can we disable ingress sync by default?

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@ishankhare07 thanks for the PR! Looks great so far, but would be great if we could default ingresses to false from here on. Please also include some unit tests.

File: cmd/vcluster/context/context.go [hunk 2]
```diff
@@ -225,6 +225,14 @@ func parseControllers(options *VirtualClusterOptions) (map[string]bool, error) {
 		}
 	}
 
+	// enable ingressclasses if ingress syncing is enabled
+	// and ingressclasses is not explicitly disabled
+	if _, ok := enabledControllers["ingresses"]; ok {
+		if _, ok = disabledControllers["ingressclasses"]; !ok {
+			enabledControllers["ingressclasses"] = true
+		}
+	}
```
Comment (FabianKramm) [inline cmd/vcluster/context/context.go:234]: Remove this

File: charts/k8s/values.yaml [hunk 3]
```diff
@@ -40,7 +40,9 @@ sync:
   persistentvolumeclaims:
     enabled: true
   ingresses:
-    enabled: true
+    enabled: false
+  ingressclasses:
```
Comment (FabianKramm) [inline charts/k8s/values.yaml:44]: Remove this

File: charts/k8s/templates/rbac/clusterrole.yaml [hunk 4]
```diff
@@ -32,6 +32,11 @@ rules:
     resources: ["storageclasses"]
     verbs: ["get", "watch", "list"]
   {{- end }}
+  {{- if or .Values.sync.ingressclasses.enabled .Values.sync.ingresses.enabled }}
```
Comment (FabianKramm) [inline charts/k8s/templates/rbac/clusterrole.yaml:35]: Remove `or .Values.sync.ingressclasses.enabled`

**Review by FabianKramm — CHANGES_REQUESTED** [review]

**Review by FabianKramm — APPROVED** [review]

File: charts/k3s/templates/rbac/clusterrole.yaml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-{{- if or (not (empty (include "vcluster.serviceMapping.fromHost" . ))) (not (empty (include "vcluster.plugin.clusterRoleExtraRules" . ))) .Values.rbac.clusterRole.create (index .Values.sync "legacy-storageclasses" "enabled") .Values.sync.nodes.enabled .Values.sync.persistentvolumes.enabled .Values.sync.storageclasses.enabled .Values.sync.priorityclasses.enabled .Values.sync.volumesnapshots.enabled -}}
+{{- if or (not (empty (include "vcluster.serviceMapping.fromHost" . ))) (not (empty (include "vcluster.plugin.clusterRoleExtraRules" . ))) .Values.rbac.clusterRole.create (index .Values.sync "legacy-storageclasses" "enabled") .Values.sync.ingresses.enabled .Values.sync.nodes.enabled .Values.sync.persistentvolumes.enabled .Values.sync.storageclasses.enabled .Values.sync.priorityclasses.enabled .Values.sync.volumesnapshots.enabled -}}
```
Comment (MPV) [inline charts/k3s/templates/rbac/clusterrole.yaml:1]: Is this change missing an equivalent in the `clusterrolebinding.yaml` file as well?

...as discovered here: https://github.com/loft-sh/vcluster/issues/755#issuecomment-1255774585

Comment (MPV) [reply charts/k3s/templates/rbac/clusterrole.yaml:1]: i.e. `.Values.sync.ingresses.enabled` hasn't been added on this line:
https://github.com/loft-sh/vcluster/blob/5c3f2fe2e9e0454dff2c52f79d6a0740795fca88/charts/k3s/templates/rbac/clusterrolebinding.yaml#L1

Comment (matskiv) [reply charts/k3s/templates/rbac/clusterrole.yaml:1]: @MPV yes, that is correct. Good catch! :+1: 
We would be glad if you create an issue about this, but if you are busy then no problem, we will create one later. We wouldn't mind a PR with a fix either, but again, no pressure :slightly_smiling_face:

Comment (matskiv) [reply charts/k3s/templates/rbac/clusterrole.yaml:1]: Oh, I just noticed that the problem already got reported in #765 , so no need to create an issue.

Comment (felipecrs) [reply charts/k3s/templates/rbac/clusterrole.yaml:1]: @MPV Are you going to create a PR? If not, I can do it.

Comment (matskiv) [reply charts/k3s/templates/rbac/clusterrole.yaml:1]: @felipecrs if you plan to create a PR, please leave a comment on #765 so I can assign it to you for tracking and visibility :)

**Pull request merged**